geomstats/geometry/discrete_surfaces.py

```diff
@@ -0,0 +1,687 @@
+
+            Returns
+            -------
+            Lh: array-like, shape=[n_verticesx3]
+                Mesh Laplacian operator of the triangulated surface applied
+                 to one its tangent vector h.
+            """
+            hdiff = h[idx[0]] - h[idx[1]]
+            values = gs.stack([gs.flatten(cot)] * 3, axis=1) * hdiff
+            Lh = gs.zeros((n_vertices, 3))
+            Lh[:, 0] = Lh[:, 0].scatter_add(0, idx[1, :], values[:, 0])
+            Lh[:, 1] = Lh[:, 1].scatter_add(0, idx[1, :], values[:, 1])
+            Lh[:, 2] = Lh[:, 2].scatter_add(0, idx[1, :], values[:, 2])
+            return Lh
+
+        return L
+
+    def normals(self, point):
+        """Compute normals at each face of a triangulated surface.
+
+        Normals are the cross products between edges of each face
+        that are incident to its x-coordinate.
+
+        Parameters
+        ----------
+        point : array-like, shape=[n_vertices, 3]
+            Surface, i.e. the vertices of its triangulation.
+
+        Returns
+        -------
+        N : array-like, shape=[n_facesx1]
+            Normals of each face of the mesh.
+        """
+        V0, V1, V2 = (
+            gs.take(point, indices=self.faces[:, 0], axis=0),
+            gs.take(point, indices=self.faces[:, 1], axis=0),
+            gs.take(point, indices=self.faces[:, 2], axis=0),
+        )
+        N = 0.5 * gs.cross(V1 - V0, V2 - V0)
+        return N
+
+    def surface_one_forms(self, point):
+        """Compute the vector valued one-forms.
+
+        The one forms are evaluated at the faces of a triangulated surface.
+
+        Parameters
+        ----------
+        point :  array-like, shape=[n_vertices, 3]
+             One surface, i.e. the vertices of its triangulation.
+
+        Returns
+        -------
+        alpha : array-like, shape=[n_faces, 3, 2]
+            One form evaluated at each face of the triangulated surface.
+        """
+        # TODO: have ambient dimension be last.
+        V0, V1, V2 = (
+            gs.take(point, indices=self.faces[:, 0], axis=0),
+            gs.take(point, indices=self.faces[:, 1], axis=0),
+            gs.take(point, indices=self.faces[:, 2], axis=0),
+        )
+        return gs.stack([V1 - V0, V2 - V0], axis=1)
+
+    def face_areas(self, point):
+        """Compute the areas for each face of a triangulated surface.
+
+        The corresponds to the volume area for the surface metric, that is
+        the volume area of the pullback metric of the immersion defining the
+        surface metric.
+
+        Parameters
+        ----------
+        point :  array-like, shape=[n_vertices, 3]
+            One surface, i.e. the vertices of its triangulation.
+
+        Returns
+        -------
+        _ :  array-like, shape=[n_faces,]
+            Area computed at each face of the triangulated surface.
+        """
+        g = self.surface_metric_matrices(point)
```

Comment:
Rename g--> surface_metrics

Reply:
are there multiple surface metrics? or should it be surface_metric (singular)?

Reply:
resolved : it should be surface_metrics plural